extensions/podman/src/podman-install.ts

```diff
@@ -0,0 +1,202 @@
+  body: string;
+  name: string;
+  assets: PodmanAsset[];
+}
+
+export async function installBundledPodman(): Promise<boolean> {
+  if (isWindows) {
+    return installWinBundledPodman();
+  } else if (isMac) {
+    return installMacBundledPodman();
+  } else {
+    //TODO: support podman installation on Linux OS
+    return false;
+  }
+}
+
+export async function installPodman(release: PodmanRelease): Promise<boolean> {
+  if (isWindows) {
+    return installWinPodman(release);
+  } else if (isMac) {
+    // return installMacPodman(release);
+  } else {
+    //TODO: support podman installation on Linux OS
+    return false;
+  }
+}
+
+export function getBundledPodmanVersion(): string {
+  return podmanTool.version;
+}
+
+export async function fetchLatestPodmanVersion(): Promise<PodmanRelease | undefined> {
+  const releases = await fetchLatestReleases();
+  return getLatestV4Release(releases);
+}
+
+function getAssetsFolder(): string {
+  const dirName = __dirname;
+  // this is temporary, as process.resourcesPath is not available in preload scripts
+  if (dirName.includes('app.asar')) {
+    const resourcesPath = dirName.substring(0, dirName.indexOf('app.asar'));
+    return path.resolve(resourcesPath, 'extensions', 'podman', 'builtin', 'podman.cdix', 'assets');
+  } else {
+    return path.resolve(__dirname, '..', 'assets');
+  }
+}
+
+async function installWinBundledPodman(): Promise<boolean> {
+  return extensionApi.window.withProgress(async progress => {
+    progress.report({ increment: 5 });
+    const msiPath = path.resolve(getAssetsFolder(), `podman-v${podmanTool.version}.msi`);
+    try {
+      if (fs.existsSync(msiPath)) {
+        await executeInstaller('msiexec', ['/i', msiPath, '/qb', '/norestart']);
+        progress.report({ increment: 80 });
+        await extensionApi.window.showDialog('info', 'Podman', 'Podman is successfully installed.', 'OK');
+        return true;
+      } else {
+        throw new Error(`Can't find Podman msi package! Path: ${msiPath} doesn't exists.`);
+      }
+    } catch (err) {
+      console.error('Error during install!');
+      console.error(err);
+      await extensionApi.window.showDialog(
+        'error',
+        'Podman Error',
+        'Unexpected error, during Podman installation: ' + err,
+        'OK',
+      );
+      return false;
+    } finally {
+      progress.report({ increment: -1 });
+    }
+  });
+}
+
+async function installWinPodman(release: PodmanRelease): Promise<boolean> {
+  return extensionApi.window.withProgress(async progress => {
+    progress.report({ increment: 5 });
+    const latestMsi = release.assets.find(it => it.name.endsWith('.msi'));
+    if (!latestMsi) {
+      progress.report({ increment: -1 });
+      throw new Error("Can't find latest MSI bundle");
+    }
+
+    const msiPath = path.resolve(os.tmpdir(), latestMsi.name);
+    try {
+      progress.report({ increment: 15 });
+      await downloadFile(latestMsi.browser_download_url, msiPath);
```

Comment:
is any integrity check performed?
because if the install fails such as failed download, there is no way to handle this properly from the code flow.
we have seen these situation happen.

perhaps add an issue to handle test/check for this in future?